diff --git a/android/src/main/java/expo/modules/ui/ModifierRegistry.kt b/android/src/main/java/expo/modules/ui/ModifierRegistry.kt
index c516cbefc8319443b41c8526453d47730eb74d57..0595f468c350df6e37a4c39db2f4021b8dcf1015 100644
--- a/android/src/main/java/expo/modules/ui/ModifierRegistry.kt
+++ b/android/src/main/java/expo/modules/ui/ModifierRegistry.kt
@@ -25,6 +25,8 @@ import androidx.compose.foundation.selection.selectable
 import androidx.compose.foundation.selection.selectableGroup
 import androidx.compose.foundation.selection.toggleable
 import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
 import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.foundation.shape.CutCornerShape
 import androidx.compose.foundation.shape.RoundedCornerShape
@@ -65,6 +67,10 @@ typealias ModifierFactory = @Composable (ModifierType, ComposableScope?, AppCont
 
 // region Modifier Params
 
+internal data class ContentDescriptionParams(
+  @Field val contentDescription: String = ""
+) : Record
+
 internal data class PaddingAllParams(
   @Field val all: Int = 0
 ) : Record
@@ -507,6 +513,11 @@ object ModifierRegistry {
     }
 
     // Utility modifiers
+    register("contentDescription") { map, _, _, _ ->
+      val params = recordFromMap<ContentDescriptionParams>(map)
+      Modifier.semantics { contentDescription = params.contentDescription }
+    }
+
     register("testID") { map, _, _, _ ->
       val params = recordFromMap<TestIDParams>(map)
       params.testID?.let { testID ->
diff --git a/ios/ColorPickerView.swift b/ios/ColorPickerView.swift
index 4f2ba12e7d1cac27f8da6b76e45f519b045703d2..0f639aa0fa2ffedfe5f4bf894a3801db8d3ea68c 100644
--- a/ios/ColorPickerView.swift
+++ b/ios/ColorPickerView.swift
@@ -49,7 +49,7 @@ struct ColorPickerView: ExpoSwiftUI.View {
       components.count > 1 ? components[1] : components[0],
       components.count > 2 ? components[2] : components[0],
       newColor.cgColor.alpha
-    ].map { Int(max(0, min(255, $0 * 255))) }
+    ].map { Int(max(0, min(255, ($0 * 255).rounded()))) }
 
     let format = supportsOpacity ? "#%02X%02X%02X%02X" : "#%02X%02X%02X"
     return String(format: format, rgba[0], rgba[1], rgba[2], supportsOpacity ? rgba[3] : 255)
