diff --git a/ios/RNSScreenStackHeaderConfig.mm b/ios/RNSScreenStackHeaderConfig.mm
index 3270127a09a05af656566009ee6c4437590bb435..c4d8d6d1180af91f29362e756d7b028b2586733d 100644
--- a/ios/RNSScreenStackHeaderConfig.mm
+++ b/ios/RNSScreenStackHeaderConfig.mm
@@ -673,7 +673,6 @@ RNS_IGNORE_SUPER_CALL_END
 #if !TARGET_OS_TV
           RNSSearchBar *searchBar = subview.subviews[0];
           searchBarPresent = true;
-          navitem.searchController = searchBar.controller;
           navitem.hidesSearchBarWhenScrolling = searchBar.hideWhenScrolling;
 #if RNS_IPHONE_OS_VERSION_AVAILABLE(16_0)
           if (@available(iOS 16.0, *)) {
@@ -699,6 +698,7 @@ RNS_IGNORE_SUPER_CALL_END
             }
           }
 #endif /* Check for iOS 26.0 */
+          navitem.searchController = searchBar.controller;
 #endif /* !TARGET_OS_TV */
         }
         break;
